### ISSUE TITLE
Exported BibLatex File causes warning about spurious commas

### ISSUE THREAD

**Issue by benblamey** [ZotPlus/zotero-better-bibtex#344]
Bib entries sometimes have a sole author (such as a corporation).
BBT introduces a trailing comma when exporting such entries, which generates a warning during build.

<pre>
[22711] Utils.pm:146> WARN - BibTeX subsystem: warning: comma(s) at end of name (removing)
[22712] Utils.pm:146> WARN - BibTeX subsystem: author, warning: comma(s) at end of name (removing)
</pre>

For large documents (e.g. theses) this generates a significant number of warnings, which clutters the output and hinders debugging.

versions:
Biber 2.1
biblatex 2015/04/19 v3.0 programmable bibliographies (PK/JW/AB)

The attached gist contains the zotero RDF and BBT output for an example exhibiting the issue:
https://gist.github.com/benblamey/bff50ef267598d2d51a7

Perhaps the issue relates to the Zotero representation of entries with a single-name author? (it seems to be labelled as "last name")

**Comment by retorquere** [comment]
When I export that entry using Better BibLaTeX, I get

```
@article{cozy.io,
  title = {Understand the Cozy Architecture},
  url = {http://cozy.io/hack/getting-started/architecture-overview.html},
  abstract = {bibtex: CozyArch 
bibtex[owner=Ben]},
  timestamp = {2015-09-11 22:29:59},
  author = {{cozy.io}}
}
```

I don't understand how you could have gotten the `owner = {Ben}` field in the output... BBT only extracts custom metadata from the extra field, not the abstract (at least that's where that data ends up; the "Better BibTeX JSON" translator is better for debugging than the RDF translator; the latter is not lossless/roundtrippable)

If I move the abstract to the "extra" field, I still get something else than you do:
```
@article{CozyArch,
  title = {Understand the Cozy Architecture},
  url = {http://cozy.io/hack/getting-started/architecture-overview.html},
  timestamp = {2015-09-11 22:35:46},
  author = {{cozy.io}},
  owner = {Ben}
} 
```
could you do the following for me:

* Enable "extended debugging" in the BBT preferences
* Select this entry in your library and export it using Better BibLaTeX
* Select this entry in your library, right-click, and select "send ZotPlus report"
* Post the resulting ID displayed when you click through here

**Comment by benblamey** [comment]
The entry was originally imported from a .bib file (I use Jabref) -- this may explain the extra metadata. I'll send the report shortly.

**Comment by benblamey** [comment]
Report J4CHKZFC

**Comment by benblamey** [comment]
Here is the original bibtex entry (prior to import)
<pre>
@Misc{CozyArch,
  Title                    = {Understand the Cozy Architecture},
  Author                   = {cozy.io},
  Url                      = {http://cozy.io/hack/getting-started/architecture-overview.html},

  Owner                    = {Ben},
  Timestamp                = {2014.06.07}
}
</pre>

**Comment by retorquere** [comment]
Yeah, that imports cleanly, with a small adjustment I just made. Could you try resetting the cache for this one too? The effect should be immediate.

**Comment by retorquere** [comment]
In fact you shouldn't even have to reset the cache -- a change of any kind to this entry (without resetting the cache!) ought to refresh the cache for the entry. Could you please (with extended debugging on):

* Add a period to the title, press enter
* Remove the period, press enter
* Right-click and send a new report

I will be able to see if the cache updated, and if not, hopefully what error prevented it.

**Comment by retorquere** [comment]
Are you still there?

**Comment by benblamey** [comment]
Sorry for the delay.

Performed the steps above, and sent report: PHT49QZJ

**Comment by benblamey** [comment]
Is "connector mode" not supported by BBT? How do I know if I am running in connector mode?

**Comment by retorquere** [comment]
OK, the cache for that entry is cleared successfully, so that works. A new BBLT export should be what you expect. You can drop the full cache from the debug pane in the BBT prefs.

Connector mode is not supported, no; you know you're in connector mode when Both Zotero Firefox and Zotero Standalone are running on the same Firefox profile. There are known problems with this setup that I have not yet been able to fix; #143 is the tracking bug for this.

**Comment by benblamey** [comment]
Yep, works ok when the cache is cleared. Thanks!
